src/platform/Zephyr/BLEAdvertisingArbiter.h

```diff
@@ -0,0 +1,82 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <lib/support/Span.h>
+
+#include <zephyr/bluetooth/bluetooth.h>
+#include <zephyr/sys/slist.h>
+
+#include <cstdint>
+
+namespace chip {
+namespace DeviceLayer {
+
+/**
+ * @brief Bluetooth LE advertising arbiter
+ *
+ * The purpose for this class is to coordinate BLE advertising between different application
+ * components. The application components that want to advertise BLE services are expected to
+ * define a request with a desired priority, a pass it to the BLE advertising arbiter.
+ * In case multiple components request BLE advertising at the same time, the arbiter selects
+ * the one with the highest priority (represented by the lowest numeric value).
+ */
+class BLEAdvertisingArbiter
+{
+public:
+    struct Request : public sys_snode_t
+    {
+        uint8_t priority;                     ///< Advertising request priority. Lower value means higher priority
+        uint32_t options;                     ///< Advertising options: bitmask of BT_LE_ADV_OPT_XXX constants from Zephyr
+        uint16_t minInterval;                 ///< Minimum advertising interval in 0.625 ms units
+        uint16_t maxInterval;                 ///< Maximum advertising interval in 0.625 ms units
+        Span<const bt_data> advertisingData;  ///< Advertising data fields
+        Span<const bt_data> scanResponseData; ///< Scan response data fields
+    };
+
+    /**
+     * @brief Request BLE advertising
+     *
+     * Add request to the internal list. If the request has higher priority than other requests,
+     * restart the BLE advertising using parameters included in the new request immediately.
```

Comment:
I wonder about one thing. Some services like Matter care about advertisements content and included UUID, and some don't like DFU over SMP, that relies only on the GATT service discovery to connect only to proper devices. That's why we are able to connect to device over SMP if it advertises only Matter characteristics UUID. But I believe there might be cases that user would like to have both e.g. Matter and some other service data advertised at the same time. Would it be possible/wise to not advertise the data of the highest priority consumer, but append data of all services from the list?

Reply:
I thought about this, but the space for advertising data is very limited (31 bytes), so only in limited scenarios it would be possible to include service data for multiple components. I could, however, probably try to combine as much requests as possible in a single one. And add an "exclusive" flag to the request to request exclusive access if one needs that.

Reply:
If it's not possible I'm also ok with that. In such case maybe we should return some error in `RequestAdvertising` informing that there is higher priority service advertising, so a new one will not be included, instead of `CHIP_NO_ERROR`?

Reply:
Hmm. The thing in such a case the request is accepted, it will just be handled later, so I'm not sure if returning a special error code to indicate that the request is deferred is the cleanest API (everyone will be forced to do a special condition like `if (request.deferred) then ignore the error`. How about adding two optional callbacks to the request: `OnRequestActivated` and `OnRequestDeactivated` that a user will be able to set if he's interested the time of processing the request?

Reply:
> If it's not possible I'm also ok with that. In such case maybe we should return some error in RequestAdvertising informing that there is higher priority service advertising, so a new one will not be included, instead of CHIP_NO_ERROR?

Maybe it's just a name of the method that could be changed, e.g. `ScheduleAdvertising` or `QueueAdvertisingRequest` (`RequestAdvertising`might indeed suggest that the request is either handled right away or dropped).